core/internal/filetransfer/file_transfer_retry_policy.go

```diff
@@ -13,13 +14,28 @@ func FileTransferRetryPolicy(
 	resp *http.Response,
 	err error,
 ) (bool, error) {
-	// Abort on any error from the HTTP transport.
-	//
-	// This happens if reading the file fails, for example if it is a directory
-	// or if it doesn't exist. retryablehttp's default policy for this situation
-	// is to retry, which we do not want.
 	if err != nil {
-		return false, err
+		switch {
+		// Retry dial tcp <IP>: i/o timeout errors.
+		//
+		// Cloud storage providers sometimes return such i/o timeout errors when rate limiting
+		// without specifying the error type to prevent potential malicious attacks.
+		// See https://wandb.atlassian.net/browse/WB-18702 for more context.
+		case strings.Contains(err.Error(), "dial tcp") && strings.Contains(err.Error(), "i/o timeout"):
+			return true, err
+		// Retry context deadline exceeded errors.
+		//
+		// This is controlled by the _file_transfer_timeout_seconds setting.
+		case strings.Contains(err.Error(), "context deadline exceeded"):
```

Comment:
It smells, but in this case the error is just a generic *url.Error, so I guess it's kinda ok.

Reply:
I don't think we should be retrying this one. This can come from the `http` package when the request timeout is exceeded---if we're seeing this a lot, then the correct thing to do is to use a higher timeout.